@navikt/core/css/darkside/toggle-group.darkside.css

```diff
@@ -1,156 +1,106 @@
+  &[aria-checked="true"] {
+    background-color: var(--ax-bg-accent-strong-pressed);
+    color: var(--ax-text-accent-contrast);
   }
-}
 
-.navds-toggle-group--neutral > .navds-toggle-group__button:focus:hover[aria-pressed="false"] {
-  box-shadow:
-    0 0 0 1px var(--ac-toggle-group-button-neutral-hover-bg, var(--a-surface-neutral-subtle-hover)),
-    0 0 0 4px var(--a-border-focus);
-}
+  & .navds-toggle-group__button-inner {
+    display: flex;
+    align-items: center;
+    gap: var(--ax-spacing-2);
+    font-weight: inherit;
 
-@supports not selector(:focus-visible) {
-  .navds-toggle-group--neutral > .navds-toggle-group__button:focus-visible:hover[aria-pressed="false"] {
-    box-shadow:
-      0 0 0 1px var(--ac-toggle-group-button-neutral-hover-bg, var(--a-surface-neutral-subtle-hover)),
-      0 0 0 4px var(--a-border-focus);
+    > * {
+      flex-shrink: 0;
+    }
+
+    > svg {
+      font-size: var(--ax-font-size-heading-medium);
+    }
   }
 }
 
-.navds-toggle-group__button-inner {
-  display: flex;
-  align-items: center;
-  gap: var(--a-spacing-2);
-  font-weight: inherit;
-}
+/* Modifier neutral variant */
+.navds-toggle-group--neutral {
+  background-color: transparent;
+  box-shadow: inset 0 0 0 1px var(--ax-border-default);
 
-.navds-toggle-group__button-inner > * {
-  flex-shrink: 0;
-}
+  & .navds-toggle-group__button {
+    background-color: transparent;
+    color: var(--ax-text-default);
 
-.navds-toggle-group__button[aria-pressed="true"],
-.navds-toggle-group__button[aria-checked="true"] {
-  background-color: var(--ac-toggle-group-selected-bg, var(--a-surface-action-selected));
-  color: var(--ac-toggle-group-selected-text, var(--a-text-on-action));
-}
+    &:hover {
+      background-color: var(--ax-bg-neutral-moderate-hoverA);
+    }
 
-.navds-toggle-group--neutral > .navds-toggle-group__button[aria-pressed="true"],
-.navds-toggle-group--neutral > .navds-toggle-group__button[aria-checked="true"] {
-  background-color: var(--ac-toggle-group-neutral-selected-bg, var(--a-surface-neutral-selected));
-  color: var(--ac-toggle-group-neutral-selected-text, var(--a-text-on-neutral));
+    &[aria-checked="true"] {
+      background-color: var(--ax-bg-neutral-strong-pressed);
+      color: var(--ax-text-neutral-contrast);
+    }
+  }
 }
 
-.navds-toggle-group--small > .navds-toggle-group__button {
-  padding: var(--a-spacing-05) var(--a-spacing-3);
-  min-height: 1.625rem;
-  min-width: fit-content;
-}
+/* Modifier small size */
+.navds-toggle-group--small {
+  /* padding: calc(var(--ax-spacing-1)); */
 
-.navds-toggle-group__button > .navds-toggle-group__button-inner > svg {
-  font-size: 1.5rem;
-}
+  & .navds-toggle-group__button {
+    padding: var(--ax-spacing-05) var(--ax-spacing-3);
+    min-height: 1.5rem;
 
-.navds-toggle-group--small > .navds-toggle-group__button > .navds-toggle-group__button-inner > svg {
-  font-size: 1.125rem;
+    & .navds-toggle-group__button-inner > svg {
```

Comment:
I read that comment from @HalvorHaugan as "dont nest stuff just to nest stuff", but "& .abcd" is still good when nesting. But might have misunderstood so please correct me if im wront @HalvorHaugan

Reply:
I think we should include the `&` even when not necessary. It makes it easier to see that it's nested. Indentation can be hard to see sometimes.

(Also, some old browsers requires it, but I guess that doesn't matter since we "un-nest" it on build anyways.)

Reply:
But I also think we should avoid unnecessary nesting. For example, `.navds-toggle-group__button-inner` shouldn't need to be nested under `.navds-toggle-group__button`, but should rather be a sibling. Nesting adds some mental "challenges" since you have to remember what the `&` represents. It gets even harder with more levels of nesting.

Reply:
@JulianNymark: It was meant more like "I don't see the need for it" (as I am quite used to nesting from Sass/Less, but it is not a hard stance.

@HalvorHaugan: Does that mean that we use nesting when using child/descendant selectors or `__` in BEM, but not to break up classes? So `.navds-toggle-group__button-inner` --> not nesting, but `.navds__button svg` or `.navds__button__svg` --> nesting´? Or did you mean some other criteria/more restrictive use?

Reply:
@it-vegard Not sure if I fully understand the question. Simply put; if it doesn't matter who the parent is, then don't use nesting.

Example:
```html
<div class="my-div">
  <div class="my-div__inner">Foo</div>
</div>
```
To target the inner element, I think we should just write `.my-div__inner {}` and not  `.my-div .my-div__inner {}` or `.my-div { & .my-div__inner {} }`. Ofc. provided that the elements will always be nested like this, and the classnames are unique (which BEM ensures).

Reply:
I'm now almost in agreement with this:

> To target the inner element, I think we should just write .my-div__inner {} and not .my-div .my-div__inner {}

But I (for some strange reason) want to have 1 forced level of nesting at the top. So that the top-most class (component representation class) will neste _everything_ for that component, even if the classes used inside it are unique and could stand on their own at the top-level too. I think it just neatly groups them without being (hopefully) too much.

Variants get their own top-level tree too, along with media queries (since they can change many things dispersed throughout any tree, just like variants). Makes it easier for me to know where to drop a new class if i add a new html element to a variant. I can also put these delusions behind me too, I don't think I've thought about them enough (would likely have to try it out a bit first to see if it's silly).

Reply:
and I guess i'll add the "why" I think it's worth it: It should in theory let you scope down as much CSS as possible at a time if the CSS trees are as "large" as possible. if there's a bunch of selectors at the top level, since everything is unique, then you would in theory have to comb through the whole file or worry that they are not grouped nicely through a weaker convention? (css comment headers to section the CSS, and hope that future modifiers don't put stuff out of order).

Reply:
> But I (for some strange reason) want to have 1 forced level of nesting at the top. So that the top-most class (component representation class) will nest everything for that component, even if the classes used inside it are unique and could stand on their own at the top-level too. I think it just neatly groups them without being (hopefully) too much. Nothing should leak outside the parent class / element (as it's the singular root ReactNode for a component), and wouldn't require the dreaded "reshuffle your CSS now because you changed something in the markup/html/tsx)"

I would argue that if we have issues with leaking classes between components or having to reshuffle things, there are other ways to solve this and more an indication of systematic issues in our process/setup. 

I don't really see adding a forced nesting-level solving anything novel since it only moves the issue one level further down the stack by increasing the specificity. Inside that nesten level you will still have to care about ordering anyways. We have a single file for each component anyways, so each component is separated by domain and ordering in its current form. 

Another side-effect is as shown the added physical filesize. We currently have 1,705 selectors in our system, if we assume 1500 of these starts with `.navds` we end up increasing our css size by ~10%.